zkchannels-contract/zkchannel_smartpy_script.py

```diff
@@ -161,46 +184,52 @@ def custClose(self, custBal, merchBal, revLock, s1, s2):
         # Note that all addresses must be implicit accounts (tz1), not smart contracts
         sp.if merchBal != sp.tez(0):
             sp.send(self.data.merchAddr, merchBal)
-        self.data.merchBal = sp.tez(0)
         self.data.status = CUST_CLOSE
  
     # merchDispute can be called if the merchant has the secret corresponding
     # to the latest custClose state. If the secret is valid, the merchant will
     # receive the customer's balance too.
     @sp.entry_point
     def merchDispute(self, secret):
+        # Only allow the merchant to call the entrypoint.
         sp.verify(self.data.merchAddr == sp.sender)
+        # Verify that the contract status is CUST_CLOSE.
         sp.verify(self.data.status == CUST_CLOSE)
         
-        # convert rev_lock in storage from LE to BE
+        # Convert rev_lock in storage from LE to BE
         revlock_be = sp.local('revlock_be', sp.list([]))
         sp.for i in sp.range(0, 32):
             revlock_be.value.push(sp.slice(self.data.revLock, i, 1).open_some())
-            
+        # Verify the revocation secret hashes to the revocation lock
         sp.verify(sp.concat(revlock_be.value) == sp.sha3(secret))
+        # Send the customer's (revoked) balance to the merchant
         sp.send(self.data.merchAddr, self.data.custBal)
-        self.data.custBal = sp.tez(0)
         self.data.status = CLOSED
  
     # custClaim can be called by the customer to claim their balance, but only
     # after the delay period from custClose has expired.
     @sp.entry_point
     def custClaim(self):
+        # Only allow the customer to call the entrypoint.
         sp.verify(self.data.custAddr == sp.sender)
+        # Verify that the contract status is CUST_CLOSE.
         sp.verify(self.data.status == CUST_CLOSE)
+        # Verify that the delay period has passed.
         sp.verify(self.data.delayExpiry < sp.now)
+        # Send the customer's balance to the customer.
         sp.send(self.data.custAddr, self.data.custBal)
-        self.data.custBal = sp.tez(0)
         self.data.status = CLOSED
  
-    # mutualClose can be called by either the customer or the merchant and
-    # allows for an instant withdrawal of the funds. mutualClose requires
-    # a signature from the merchant and the customer on the final state.
+    # mutualClose can only be called by the customer and allows for an instant withdrawal
+    # of the funds. mutualClose requires a signature from the merchant over tuple containing
```

Comment:
missing "a" before "tuple"

Reply:
clarify what type of signature.

Reply:
added "a"
added that it's an 'EdDSA signature'